src/integrations/kafka/consumer.cpp

```diff
@@ -336,16 +340,24 @@ void Consumer::StartConsuming() {
       if (batch.empty()) continue;
 
       spdlog::info("Kafka consumer {} is processing a batch", info_.consumer_name);
-
-      try {
-        consumer_function_(batch);
-        if (const auto err = consumer_->commitSync(); err != RdKafka::ERR_NO_ERROR) {
-          spdlog::warn("Committing offset of consumer {} failed: {}", info_.consumer_name, RdKafka::err2str(err));
+      constexpr size_t max_retries = 30;
+      // to-do kostas make this configurable
+      for (size_t i = 0; i < max_retries; ++i) {
+        try {
+          consumer_function_(batch);
+          if (const auto err = consumer_->commitSync(); err != RdKafka::ERR_NO_ERROR) {
+            spdlog::warn("Committing offset of consumer {} failed: {}", info_.consumer_name, RdKafka::err2str(err));
+          }
           break;
+        } catch (const std::exception &e) {
+          const std::string_view error(e.what());
+          spdlog::warn("Error happened in consumer {} while processing a batch: {}!", info_.consumer_name, error);
+          if (error != query::kSerializationErrorMessage) {
+            break;
+          }
+          using namespace std::chrono_literals;
+          std::this_thread::sleep_for(500ms);
```

Comment:
This shouldn't be here. Please modify in `CreateConsumer` in `streams.cpp` the way queries are being run. We want this to work for all consumers not just for Kafka.
Wrapping `interpreter->PullAll` in try/catch should be enough. 
Another reason not to do this is using something from the query lib in the integrations lib. `integrations` lib should be standalone.